private-templates-service/client/src/components/Common/VersionModal/VersionModal.tsx

```diff
@@ -0,0 +1,159 @@
+
+  unpublish = () => {
+    for (let instance of this.props.template.instances!){
+      if (!this.state.versionList.includes(instance.version!)) continue;
+      const template = JSON.stringify(instance.json);
+      this.props.updateTemplateState(template, PostedTemplate.StateEnum.Deprecated, instance.version!);
+      this.props.toggleModal();
+    }
+  }
+
+  addVersion = (isChecked?: boolean, version?: string) => {
+    if (!isChecked || !version) return;
+    let versionList = this.state.versionList;
+    if (isChecked === true) {
+      if (versionList.includes(version)) return;
+      versionList.push(version);
+    } else {
+      if (!version.includes(version)) return;
+      let index = versionList.indexOf(version);
+      versionList.splice(index, 1);
+    }
+    this.setState({ versionList: versionList });
+  }
+
+  render() {
+    const { template } = this.props;
+    return (
+      <BackDrop>
+        <Modal>
+          <Header>Versions</Header>
+          <Description>There are {template.instances!.length} versions for <DescriptionAccent>{template.name}</DescriptionAccent></Description>
+          <CenterPanelWrapper>
+            <Card>
+              <CardHeaderRow>
+                <CardHeaderText>Version</CardHeaderText>
+                <CardHeaderText>Published</CardHeaderText>
+                <CardHeaderText>Status</CardHeaderText>
+                <CardHeaderText>{`${this.state.versionList!.length} Selected`}</CardHeaderText>
+              </CardHeaderRow>
+              <CardBody>
+              {this.props.template.instances && this.props.template.instances.map((instance: TemplateInstance) => (
+            <VersionCardRow>
+              <VersionWrapper>
+                {instance.version}
+              </VersionWrapper>      
+              <DateWrapper>{instance.publishedAt? getDateString(instance.publishedAt) : "Not published"}</DateWrapper>
+              <StatusWrapper>
+                <StatusIndicator state={instance.state}/>
+                <Status>{instance.state && instance.state.toString().charAt(0).toUpperCase() + instance.state.toString().slice(1)}</Status>
+              </StatusWrapper>
+              <CheckboxWrapper><Checkbox defaultChecked={false} onChange={(e, checked) => this.addVersion(checked, instance.version)}/></CheckboxWrapper>
+            </VersionCardRow>
+            ))}     
+              </CardBody>
+            </Card>
+          </CenterPanelWrapper>
+          <BottomRow>
+            <ButtonGroup>
+              <LightButton text="Cancel" onClick={this.props.toggleModal} />
+              <LightButton text="Delete" onClick={this.delete} />
+              <PrimaryStyleButton text="Unpublish" onClick={this.unpublish} />
+              <PrimaryStyleButton text="Publish" onClick={this.publish} />
+            </ButtonGroup>
+          </BottomRow>
+        </Modal>
+      </BackDrop>
+    );
+  }
+}
+
+export default connect(() => { return {} }, mapDispatchToProps)(VersionModal);
```

Comment:
Can you wrap this in the `ModalHOC` ?

Reply:
See `PublishModal`
![image](https://user-images.githubusercontent.com/19780117/75935398-f087dc00-5e33-11ea-8644-56a897f60f56.png)

Reply:
If you can update the other modal component you merged, that would be great.

Reply:
Fixed, thanks!